src/Model/Post/Media.php

```diff
@@ -571,6 +576,63 @@ public static function getType(string $mimeType): int
 		return $type;
 	}
 
+	/**
+	 * Fetch video dimensions using getID3
+	 *
+	 * @param array $media
+	 * @return array media with added dimensions
+	 */
+	private static function getVideoDimensions(array $media): array
+	{
+		if (!empty($media['width']) && !empty($media['height'])) {
```

Comment:
@art4 - What do you think about `empty()` usages here? shouldn't there be a lot more `isset()` calls instead?

tbh I really don't know what check is for which situation better ;-) ;-=)

Reply:
The entries can already be there, but when they aren't filled with a number larger than `0` we have to fetch it on our own.

Reply:
So in case we do want to avoid `empty()` (as suggested from @art4 and found at best practices like https://dev.to/klnjmm/never-use-empty-function-in-php-4pb0)
```suggestion
		if (isset($media['width']) && $media['width'] > 0 && 
		    isset($media['height']) && %media['isset'] > 0) {
```

Reply:
@nupplaphil 👍 for your suggestion. But we also have to add an explicit casting to `int` while setting `$media['width']` and `$media['heigth']`.

Reply:
In this context, I guess there are some things that I will never understand, such as the fact that PHP is much more capable than the language I earn my money with, yet people refrain from using its full potential for whatever reason. I can apply the changes, but I don't really understand what's so bad about using `empty` when we really want to check if the variable is set *and* contains a value that is not `0` (or an empty string, ...)

Reply:
I'll try to explain it in this context: As you said the entries can already be there, but atm we have not control what type the values could be. If for some weird reason the value contains an non-empty array instead of an int, `empty()` will return `true` and this code maybe (or maybe not) will break at a later state. The chances are also high that such a bug will not be detected but will lead to strange behavior/bugs in the future.

Instead, at this point in the code we are really sure, that no arrays/objects/strings/bool/etc are allowed as height/width value, because we know for sure it must be either an integer greater than 0 or not set at all. So we should check `isset()` and `> 0` here. If now someone somewhere else (accidentally) tries to set the width/height values, their code wont work and will tell the developer what type the height/width values must be.

The usage of `empty()` might have some valid use cases in small scripts, but in large or long-lived codebases this is an anti-pattern.

Reply:
What about creating a helper class for typical cast-checks, which would help using it the right way?

Like
```php
/**
* Use it in case a value you pass must be a valid integer and greater zero.
*/
Checker::isPositiveInt(int $value): bool
{
  return $value > 0:
}
```

Thus we avoid explicit type-casting and the check has a context when using it. Maybe this helps using it more often :)

Reply:
The underlying issue is the [primitive obsession code smell](https://refactoring.guru/smells/primitive-obsession). Instead of cementing the status quo, we should rather switch `$media` into a value object in the medium term.

Reply:
Puuuuhhh.. I know what you mean, but I find it very hard to introduce it as a short-fix. We have to work through the whole code and firsthand we have to enable it in a way that we don't loose Devs (foremost @annando as he said it's a different world for him)

It's three steps ahead I think, shouldn't we introduce a primitive check-class at first with the hint of using type safe implementations...

Reply:
I still need to really understand, what "smells" are. I guess it stands for a bad code practice? But on that page I only see some: "This is bad" and not "this is bad, because ...". Also I currently have got a problem to understand how this "smell" refers to the code.

Reply:
I now changed the code to avoid `empty`.

Reply:
> I still need to really understand, what "smells" are. I guess it stands for a bad code practice? But on that page I only see some: "This is bad" and not "this is bad, because ...". Also I currently have got a problem to understand how this "smell" refers to the code.

code smells are bad practices, which leads to harder to maintain code and more vulnerable for bugs because of unseen flaws behind code.

Like @art4 said, `empty()` doesn't guard the code from mis-formatted values in an array, like string values instead of int. That may lead to later regressions, which are harder to tackle down since the root cause could be much earlier in the execution-process, but flies under the radar.

The best way would be to use data in specific, type-safe objects, so there are no worries about casting/checking anymore. But that's a paradigm-change in the current way we created the whole federation classes. We simply would avoid using any named-array in favour of data-objects. And move data integrity checks of data INSIDE the object class and using the data OUTSIDE.

All the "is the array filled with correct data, if not do something else and fill other arrays with data" would vanish. The federation processor would always get correct data to process and leave all integrity checks to that classes.

It's like creating a car class which ensures there are 4 tiers and the processor just calls "drive away" instead of "check if the car has tiers and if the engine exists at if so, drive away, else buy new tiers, add them and then drive away"